src/main/java/com/mongodb/jdbc/SortableBsonDocument.java

```diff
@@ -0,0 +1,61 @@
+package com.mongodb.jdbc;
+
+import java.util.List;
+import org.bson.BsonDocument;
+
+public class SortableBsonDocument extends BsonDocument implements Comparable<SortableBsonDocument> {
+
+    static class SortSpec {
+        String key;
+        ValueType type;
+
+        SortSpec(String key, ValueType type) {
+            this.key = key;
+            this.type = type;
+        }
+    }
+
+    enum ValueType {
+        String,
+        Int,
+    }
+
+    List<SortSpec> sortSpecs;
+    BsonDocument nestedDocValue;
+
+    SortableBsonDocument(List<SortSpec> sortSpecs, String key, BsonDocument docValue) {
```

Comment:
Note that this is exclusively constructed as `new SortableBsonDocument(<sortSpecs>, BOT_NAME, botDoc)`, and since we compare values nested in the `botDoc`, I just store that value to avoid additional `this.getDocument(BOT_NAME)` calls in the `compareTo` method.